Library/Homebrew/test/cask/artifact/app_spec.rb

```diff
@@ -334,5 +334,29 @@
 
       expect(contents_path).to exist
     end
+
+    it "properly handles non-writable directories" do
+      install_phase
+
+      contents_path = target_path.join("Contents")
+
+      expect(app.target).to receive(:writable?).at_least(:once).and_return false
+      allow(command).to receive(:run!).and_call_original
+      allow(command).to receive(:run!)
+        .with("/bin/cp", args: ["-pR", source_path.join("Contents"), contents_path],
+                         sudo: true)
+        .and_wrap_original do |original_method, *args, **kwargs|
+          original_method.call(*args, **kwargs, sudo: false)
+        end
```

Comment:
Instead of this, I think we can change `SystemCommand` above to `NeverSudoSystemCommand`.

Reply:
Done.